EightfoldAI/octuple
# fix: search box: pressing enter key clears text when clearable is true

## SUMMARY:
Handle clear button keyboard event so it doesn't trigger by default. It now requires focus.

Also updates buttons to newer system ui button with transparency. 

## JIRA TASK (Eightfold Employees Only):
ENG-32974

## CHANGE TYPE:


## TEST COVERAGE:


## TEST PLAN:

- Pull the PR branch and run `yarn` and `yarn storybook`. Verify the SearchBox and TextInput stories behave as expected.
- Open the ConfigProvider story and tab around to activate keyboard styles
- Go to the SearchBox story and click on the SearchBox and type some text. Pressing enter should trigger a bunko search, not clear the text. Tabbing to the clear icon and pressing enter should clear the text.
